### DISCUSSION TITLE
403 error on flows

### DISCUSSION THREAD

**Discussion by nordcart** [directus/directus#15017] [Archive]
### Describe the Bug

I´m using the manual trigger to get ID:s of selected items of a collection {{$trigger.body.keys}}
When using read data to get information on the selected items, I´m getting a 403 error.
When {{$trigger.body.keys[0]}} is used it works fine, but I´m only getting information from the first item.

### To Reproduce

Read data IDs as: {{$trigger.body.keys}}

### Errors Shown

![image](https://user-images.githubusercontent.com/12813484/184092688-078c7080-d86c-45d8-8163-4f1998794984.png)

### What version of Directus are you using?

9.15.1

### What version of Node.js are you using?

dont know

### What database are you using?

Postgres 12

### What browser are you using?

Chrome

### How are you deploying Directus?

Docker

**Comment by azrikahar** [comment]
It seems to work from my testing:

![image](https://user-images.githubusercontent.com/42867097/184097350-0e4cab3d-5cba-4813-bde6-600ba00fcb3a.png)

![image](https://user-images.githubusercontent.com/42867097/184097457-04dc50c6-7648-47dd-8fc4-3f2683a4a5e3.png)

However it does seem odd that your Read Data's key is being escaped. Can you share how did you enter `{{$trigger.body.keys}}`? Here's what I'm using in my setup:

![image](https://user-images.githubusercontent.com/42867097/184097673-0da365c7-3f3e-4345-bf7d-aa3e2c4f88f8.png)

Moving this to discussion for now 👍

**Reply by nordcart** [reply]
![image](https://user-images.githubusercontent.com/12813484/184102664-71612248-ce04-4130-b251-7478f424ab06.png)

I pressed this and noticed "[ ]" I removed those and it is working now, Don´t know where they came from...

**Reply by azrikahar** [reply]
> ![image](https://user-images.githubusercontent.com/12813484/184102664-71612248-ce04-4130-b251-7478f424ab06.png)
> 
> I pressed this and noticed "[ ]" I removed those and it is working now, Don´t know where they came from...

Ohh I believe it's due to entering `{{$trigger.body.keys}}` in IDs, _without_ using the `{}` raw variables editor! so the tags interface is turning it into an array. Glad it's resolved!
